titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -653,6 +620,49 @@ private void podUpdated(V1Pod pod) {
         }
     }
 
+    private void podDeleted(V1Pod pod) {
+        try {
+            V1ObjectMeta metadata = pod.getMetadata();
+            V1PodStatus status = pod.getStatus();
+            String podName = KubeUtil.getMetadataName(metadata);
+
+            if (!isPodValid(pod, metadata, status, podName)) {
```

Comment:
if a Pod is deleted (by anyone) and it didn't have a proper status field set yet, we wait for full reconciliation to catch it?

Shouldn't we consider emitting a `ContainerEvent` with `REASON_TASK_LOST` (or `REASON_UNKNOWN`) here, for that case?